tests/test_inference_client.py

```diff
@@ -1080,3 +1084,121 @@ def test_resolve_chat_completion_url(
     client = InferenceClient(model=client_model, base_url=client_base_url)
     url = client._resolve_chat_completion_url(model)
     assert url == expected_url
+
+
+@pytest.mark.parametrize(
+    "inputs, parameters, expect_binary, expected_json, expected_data",
+    [
+        # Case 1: inputs is a simple string without parameters
+        (
+            "simple text",
+            None,
+            False,
+            {"inputs": "simple text"},
+            None,
+        ),
+        # Case 2: inputs is a simple string with parameters
+        (
+            "simple text",
+            {"param1": "value1"},
+            False,
+            {
+                "inputs": "simple text",
+                "parameters": {"param1": "value1"},
+            },
+            None,
+        ),
+        # Case 3: inputs is a dict without parameters
+        (
+            {"input_key": "input_value"},
+            None,
+            False,
+            {"inputs": {"input_key": "input_value"}},
+            None,
+        ),
+        # Case 4: inputs is a dict with parameters
+        (
+            {"input_key": "input_value", "input_key2": "input_value2"},
+            {"param1": "value1"},
+            False,
+            {
+                "inputs": {"input_key": "input_value", "input_key2": "input_value2"},
+                "parameters": {"param1": "value1"},
+            },
+            None,
+        ),
+        # Case 5: inputs is bytes without parameters
+        (
+            b"binary data",
+            None,
+            True,
+            None,
+            b"binary data",
+        ),
+        # Case 6: inputs is bytes with parameters
+        (
+            b"binary data",
+            {"param1": "value1"},
+            True,
+            {
+                "inputs": "encoded_data",
+                "parameters": {"param1": "value1"},
+            },
+            None,
+        ),
+        # Case 7: inputs is a Path object without parameters
+        (
+            Path("test_file.txt"),
+            None,
+            True,
+            None,
+            Path("test_file.txt"),
+        ),
+        # Case 8: inputs is a Path object with parameters
+        (
+            Path("test_file.txt"),
+            {"param1": "value1"},
+            True,
+            {
+                "inputs": "encoded_data",
+                "parameters": {"param1": "value1"},
+            },
+            None,
+        ),
+        # Case 9: inputs is a URL string without parameters
+        (
+            "http://example.com",
+            None,
+            True,
```

Comment:
Would be good to test URL + `expect_binary = False` => shouldn't be encoded. (cf comment above).
Same for case 9 and 10.